### ISSUE TITLE
Generative Models as a Data Source for Multiview Representation Learning

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#200]
# TL; DR

- Using generative models with latent space transformation produces a reasonable alternative for contrastive representation learning with real dataset.

# Motivation

- Replacing real dataset with a generative model has several advantages: compression, privacy concern and controllability (e.g., in terms of bias).

# Formulation

<img width="966" alt="image" src="https://user-images.githubusercontent.com/46308407/207538202-0b8f4474-d4f8-4ef1-be63-38ee0d878fd8.png">
<img width="967" alt="image" src="https://user-images.githubusercontent.com/46308407/207538222-f1422ab4-521e-4337-8986-d8eb87299472.png">

- The authors aim to learn an embedding function (denoted **F**) from either a synthetic dataset that is the output of an off-the-shelf generative model.
- As multiview representation learning relies on semantic-invariant transformation (i.e., data augmentation), the authors suggest latent space transformation.

<img width="762" alt="image" src="https://user-images.githubusercontent.com/46308407/207538588-74654a19-aea0-4836-a24a-08892d881471.png">

- There can be several design choices as to how to obtain the positive sample in the latent space.
- One is simply adding a Gaussian noise (with moderate s.d.), and the other is steering it towards a meaningful semantic direction (as done in GAN-inversion-based image editing models).
- Aside from the contrastive frameworks, the authors introduce an inverter baseline, where the representation learner is simply trained to invert the original latent in the generative latent space (with L2 loss).

# Evaluation

<img width="955" alt="image" src="https://user-images.githubusercontent.com/46308407/207544176-aa5c2b90-8abb-49a1-8517-a53f31fc2556.png">

- Augmentation in the latent space **and** the pixel space combined yields the best performance, but overall falls short to real data training (as expected, I would say).

<img width="909" alt="image" src="https://user-images.githubusercontent.com/46308407/207544370-4c5f5099-7b26-44e0-8189-87d5dda4e5a8.png">

- There is a clear sweet spot in the standard deviation for positive sample generation, and the dataset size scales sub-logarithmically (as the implicit diversity of a generative model is limited).

# Inspiration

- Interesting approach to incorporate latent space manipulation into the contrastive learning framework.
- The quality of the learned representation relies largely on the power of the generative model, as LSUN-Cars StyleGAN2 produces excellent samples while Imagenet is much more challenging the model with a GAN.
- Wonder if we can leverage the latent space distance for efficient dataset distillation. For example, optimizing latent codes of a generative prior instead of pixels (which would be much more scalable and manifold-complying).
